addons/account/models/account_move.py

```diff
@@ -2400,6 +2410,28 @@ def _validate_taxes_country(self):
                     raise ValidationError(_("This entry contains taxes that are not compatible with your fiscal position. Check the country set in fiscal position and in your tax configuration."))
                 raise ValidationError(_("This entry contains one or more taxes that are incompatible with your fiscal country. Check company fiscal country in the settings and tax country in taxes configuration."))
 
+    def _validate_tax_alignment_with_fpos(self):
+        """
+        Method to validate whether the tax_ids on the move_lines are aligned with the fiscal position of the move.
+
+        Returns:
+            bool: `True` if the taxes on the move_lines aligned with the fiscal position, otherwise `False`.
+        """
+
+        self.ensure_one()
+        tax_ids = self.invoice_line_ids.tax_ids
+        tax_mapping = self.env["account.fiscal.position.tax"].with_company(self.company_id).search([
+            '|',
+            ('tax_src_id', 'in', tax_ids.ids),
+            ('tax_dest_id', 'in', tax_ids.ids),
+        ])
+        if tax_mapping:
+            unmapped_tax_ids = tax_ids - (tax_mapping.tax_src_id | tax_mapping.tax_dest_id)
```

Comment:
```suggestion
            unmapped_tax_ids = self.env['account.tax'].browse(tuple(_id if _id not in set(*tax_mapping.tax_src_id.ids, *tax_mapping.tax_dest_id.ids) for _id in tax_ids.ids))
```

Reply:
Doesn't the use of `self.env['account.tax'].browse` make it more expensive if we talk about performance? Additionally, iterating over `tax_ids` and `tax_mapping` seems to increase the time complexity. Wouldn't using a union operation be more preferable? :thinking:

Reply:
do you know how does operator overloading works in the orm? I guess not you can have [look at it](https://github.com/odoo/odoo/blob/18.0/odoo/models.py#L6878) 😉 

Any operation on a recordset will return a new recordset